Cubical/Algebra/CommAlgebra/Localisation.agda

```diff
@@ -223,3 +240,76 @@ module AlgLocTwoSubsets (R' : CommRing ℓ)
     S₂⊆S₁⁻¹Rˣ s₂ s₂∈S₂ =
       transport (λ i → _⋆_ ⦃ (sym S₁⁻¹R≡S₂⁻¹R) i .snd ⦄ s₂ (1a ⦃ (sym S₁⁻¹R≡S₂⁻¹R) i .snd ⦄)
                      ∈ (CommAlgebra→CommRing ((sym S₁⁻¹R≡S₂⁻¹R) i)) ˣ) (S₂⋆1⊆S₂⁻¹Rˣ s₂ s₂∈S₂)
+
+
+
+-- A crucial result for the construction of the structure sheaf
+module DoubleAlgLoc (R : CommRing ℓ) (f g : (fst R)) where
+ open Exponentiation R
+ open InvertingElementsBase
+ open CommRingStr (snd R) hiding (·Rid)
+ open isMultClosedSubset
+ open DoubleLoc R f g hiding (R[1/fg]≡R[1/f][1/g])
+ open CommAlgChar R
+ open AlgLoc R ([_ⁿ|n≥0] R (f · g)) (powersFormMultClosedSubset R (f · g))
+             renaming (S⁻¹RAlgCharEquiv to R[1/fg]AlgCharEquiv)
+ open CommIdeal R hiding (subst-∈) renaming (_∈_ to _∈ᵢ_)
+ open isCommIdeal
+ open RadicalIdeal R
+
+ private
+  ⟨_⟩ : {n : ℕ} → FinVec (fst R) n → CommIdeal
+  ⟨ V ⟩ = ⟨ V ⟩[ R ]
+
+  R[1/fg]AsCommAlgebra = R[1/_]AsCommAlgebra {R = R} (f · g)
+  R[1/fg]ˣ = R[1/_]AsCommRing R (f · g) ˣ
+
+  R[1/g]AsCommAlgebra = R[1/_]AsCommAlgebra {R = R} g
+  R[1/g]ˣ = R[1/_]AsCommRing R g ˣ
+
+  R[1/f][1/g]AsCommRing = R[1/_]AsCommRing (R[1/_]AsCommRing R f)
+                                [ g , 1r , powersFormMultClosedSubset R f .containsOne ]
+  R[1/f][1/g]AsCommAlgebra = toCommAlg (R[1/f][1/g]AsCommRing , /1/1AsCommRingHom)
+
+ R[1/fg]≡R[1/f][1/g] : R[1/fg]AsCommAlgebra ≡ R[1/f][1/g]AsCommAlgebra
+ R[1/fg]≡R[1/f][1/g] = uaCommAlgebra (R[1/fg]AlgCharEquiv _ _ pathtoR[1/fg])
+
+ doubleLocCancel : g ∈ᵢ √ ⟨ replicateFinVec 1 f ⟩ → R[1/f][1/g]AsCommAlgebra ≡ R[1/g]AsCommAlgebra
```

Comment:
Nicer notation for `⟨ replicateFinVec 1 f ⟩` , maybe some notation for singleton vector?